libsrc/atari2600/crt0.s

```diff
@@ -0,0 +1,46 @@
+; Atari VCS 2600 startup code for cc65
+;
+; Florent Flament (contact@florentflament.com), 2017
+
+        .export         _exit
+        .export         __STARTUP__ : absolute = 1
+
+        .import         __RAM_START__, __RAM_SIZE__
+        .import         copydata
+        .import         _main
+
+        .include        "zeropage.inc"
+
+
+.segment "STARTUP"
+start:
+; Clear decimal mode
+        cld
+; Clear zeropage & registers
+; Also Initializes system stack pointer
+        ldx     #0
```

Comment:
Perhaps I am missing something obvious, but the dex / txs in the loop does not look right to me.

Why not:

ldx #0
txa
txs
clearLoop
pha
dex
bne clearLoop

I know, this is not completely the same, as I am writing the stack differently.

Even better, omitting the pha completely would save many cycles, although you would need two more byte:

ldx #0
txa
txs
clearLoop
sta $0100,x
dex
bne clearLoop